app/components/chatbox.js

```diff
@@ -0,0 +1,28 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+import io from 'socket.io/client-dist/socket.io.js';
+
+export default class ChatboxComponent extends Component {
+  socket = io();
+
+  @tracked
+  messages = [];
+
+  constructor() {
+    super(...arguments);
+
+    this.socket.on('chat message', (msg) => {
+      this.messages = [
```

Comment:
should this be guarded in:

```js
import { isDestroying, isDestroyed } from '@ember/destroyable';

// ...

if (isDestroying(this) || isDestroyed(this)) return;

this.messages = ...

```

Reply:
Lol yea probably 🤣 bit this app is incredibly simple, has exactly one route and if we're destroying this component then we're probably closing the window